Rakefile

```diff
@@ -24,15 +24,26 @@ Rake::TestTask.new(:test) do |test|
 end
 
 # Building the gem will use the local file mode, so ensure it's world-readable.
-desc 'Check plugin file permissions'
-task :check_perms do
-  plugin = 'lib/fluent/plugin/out_detect_exceptions.rb'
-  mode = File.stat(plugin).mode & 0o777
-  raise "Unexpected mode #{mode.to_s(8)} for #{plugin}" unless
-    mode & 0o444 == 0o444
+# https://github.com/GoogleCloudPlatform/fluent-plugin-detect-exceptions/issues/32
+desc 'Fix file permissions'
+task :fix_perms do
+  files = [
+    'lib/fluent/plugin/exception_detector.rb',
+    'lib/fluent/plugin/out_detect_exceptions.rb'
+  ].flat_map do |file|
```

Comment:
This `flat_map` seems to be an no-op?  It's equivalent to:

```
  files = [
    'lib/fluent/plugin/exception_detector.rb',
    'lib/fluent/plugin/out_detect_exceptions.rb'
  ]
```

Right?

Reply:
It's a good pattern to have in case we decide to add wildcards to the list of files (and it's consistent with [fluent-plugin-google-cloud/Rakefile](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/Rakefile#L22-L29)). But I'll leave it up to you.

Reply:
If we want a wild card, we can change it to:
```
  files = [
    'lib/fluent/plugin/*.rb'
  ].flat_map do |file|
    file.include?('*') ? Dir.glob(file) : [file]
  end
```

The way it is right now is a bit confusing to read.

Reply:
Yeah, the gemspec uses a wildcard to select the files to package, so your suggestion makes sense. Let's go with that, then.

Reply:
Changed